src/Native/Runtime/EHHelpers.cpp

```diff
@@ -395,8 +395,15 @@ static UIntNative UnwindWriteBarrierToCaller(
     UIntNative sp = pContext->GetSp();
     UIntNative adjustedFaultingIP = *(UIntNative *)sp;
     pContext->SetSp(sp+sizeof(UIntNative)); // pop the stack
-#elif defined(_ARM_) || defined(_ARM64_)
+#elif defined(_ARM64_)
     UIntNative adjustedFaultingIP = pContext->GetLr();
+#elif defined(_ARM_)
+    UIntNative adjustedFaultingIP = pContext->GetLr();
+    if (pContext->GetIp() == (unsigned int)RhpLockCmpXchg64AVLocation)
+    {
```

Comment:
`r4` and `r5` registers are modified as well at this point, so they would need to be restored as well.

I think it may be better to do a dummy read at the very start of the method `ldr r12, [r0]` and move the AVLocation there. It will make the method tiny bit slower, but the rest will be so much simpler.